spinta/datasets/backends/dataframe/backends/soap/components.py

```diff
@@ -4,14 +4,44 @@
 from zeep.proxy import OperationProxy
 
 from spinta.datasets.backends.dataframe.components import DaskBackend
-from spinta.utils.schema import NA
+from spinta.datasets.backends.wsdl.components import WsdlBackend
+
+from spinta.exceptions import SoapServiceError
 
 
 class Soap(DaskBackend):
     type: str = "soap"
     query_builder_type = "soap"
 
-    soap_operation: OperationProxy | NA = NA
+    _soap_operation: OperationProxy | None = None
+
+    wsdl_backend: WsdlBackend
+    service: str
+    port: str
+    operation: str
+
+    @property
+    def soap_operation(self) -> OperationProxy:
+        if not self._soap_operation:
+            self._soap_operation = self._get_soap_operation()
+
+        return self._soap_operation
```

Comment:
Shorthand could be used:
```suggestion
        return self._soap_operation if self._soap_operation else self._get_soap_operation()
```